### ISSUE TITLE
Spawn and MPI_COMM_SPAWN with the SOFT info

### ISSUE THREAD

**Issue by abouteiller** [pmix/pmix-standard#84]
It appears that there is no attribute currently defined in the PMIx_spawn operation to pass the behavior defined as a 'SOFT' spawn in MPI. 

In an MPI soft spawn, the MPI_COMM_SPAWN operation requests that N new processes be spawned, but it is acceptable that N-x processes are actually created (and this is not an error).

**Comment by rhc54** [comment]
That's a new one on me - does anyone actually support that? I know OMPI doesn't. We should be able to do so here by adding a ```PMIX_MIN_PROCS``` attribute - we already have ```max_procs``` in the ```pmix_app_t``` struct.

**Comment by dsolt** [comment]
I agree that a PMIX_MIN_PROCS option would be sufficient to support MPI_COMM_SPAWN with a soft spawn and also allow for other uses.

**Comment by jjhursey** [comment]
It sounds like there is an agreement on how this looks from the PMIx side.

Action:
 * Add a `PMIX_MIN_PROCS` attribute for `PMIx_Spawn` (similar in nature to `PMIX_MAX_PROCS`)
